test.sh

```diff
@@ -10,11 +10,11 @@
 # Exit immediately on errors
 set -e
 
-timestamp="$(date +%d%m%y%H%M%s)"
+timestamp="$(date +%y%m%d-%H%M%S)"
 # Create a temp file for docker env.
 # When the script exits (successfully or otherwise), clean up the file automatically.
-credsfile="$(mktemp /tmp/sdk-test-creds-${timestamp}.tmp)"
-envfile="$(mktemp /tmp/sdk-test-env-${timestamp}.tmp)"
+credsfile="$(mktemp /tmp/sdk-test-creds-${timestamp}-XXXX.tmp)"
```

Comment:
Just for my info, what is the motivation for the `XXXX`?

Reply:
`mktemp` uses it to insert a random token into the string:
```
$ mktemp foo-XXXX.tmp
foo-SXIq.tmp
```

Reply:
Both are useful -- sometimes you can end up with a left over file if e.g. the build is aborted, so it doesn't hurt to include an origin timestamp in there so that old ones are apparent when listing the directory.